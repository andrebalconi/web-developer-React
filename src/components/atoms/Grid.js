import styled, { css } from "styled-components";
import PropTypes from "prop-types";

import { breakAt, BreakpointSize } from "styles/Breakpoints";

const breakColumnAt = (size) => (props) =>
  props[size] &&
  css`
    ${breakAt(BreakpointSize[size])} {
      grid-template-columns: repeat(${(props) => props[size]}, 1fr);
    }
  `;

const Grid = styled.div`
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;

  ${breakColumnAt("sm")}
  ${breakColumnAt("md")}
  ${breakColumnAt("lg")}
  ${breakColumnAt("xl")}
`;

Grid.defaultProps = {
  sm: undefined,
  md: undefined,
  lg: undefined,
  xl: undefined,
};

Grid.propTypes = {
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
  xl: PropTypes.number,
};

export default Grid;
