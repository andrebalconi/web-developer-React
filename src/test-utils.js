import React from "react";
import ThemeProvider from "styles/ThemeProvider";
import { render } from "@testing-library/react";
/*import { MemoryRouter as Router } from "react-router-dom";*/

const TestProvider = ({ children }) => (
  
    <ThemeProvider>{children}</ThemeProvider>

);

export * from "@testing-library/react";

const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

export { customRender as render };
