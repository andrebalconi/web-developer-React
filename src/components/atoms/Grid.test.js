import React from "react";
import { render } from "test-utils";

import Grid from "./Grid";

test("match snapshot with no params", () => {
  const { asFragment } = render(<Grid></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("match snapshot with params", () => {
  const { asFragment } = render(<Grid sm={2} md={3} lg={4} xl={5}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("match snapshot with sm and lg", () => {
  const { asFragment } = render(<Grid sm={6} lg={12}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});
