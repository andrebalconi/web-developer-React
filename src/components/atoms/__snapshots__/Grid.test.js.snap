// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with no params 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`match snapshot with params 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`match snapshot with sm and lg 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(12,1fr);
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;
